### PR TITLE
Stop treating flux as separate service on the front end

### PR DESCRIPTION
**Pull request by ischolten**
Closes https://github.com/influxdata/applications-team-issues/issues/89

_Briefly describe your proposed changes:_
Since flux is no longer a separate service, we are not having users make a separate service connection anymore. This will take away the responsibility of the user to make a flux service connection and will be handled via the backend. If you use a source that is 1.7 version or higher, source will get a flux link added to the links section that the app will use to proxy influxdb with a flux query. 

Throughout the app now, we will only have to deal with source and won't have to pass around services everywhere in order to make request and determine whether or not flux is supported. 

  - [x] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

File: ui/src/dashboards/components/SourceSelector.tsx [hunk 1]
```diff
@@ -42,18 +38,19 @@ const SourceSelector: SFC<Props> = ({
...
     <div className="source-selector">
       <SourceDropdown
-        service={service}
-        services={services}
         source={source}
+        type={type}
         sources={sources}
         allowInfluxQL={true}
-        allowFlux={true}
+        allowFlux={sourceSupportsFlux}
```
Comment (AlirieGray) [inline ui/src/dashboards/components/SourceSelector.tsx]: nice catch

File: ui/src/dashboards/containers/DashboardPage.tsx [hunk 2]
```diff
@@ -372,21 +361,8 @@ class DashboardPage extends Component<Props, State> {
 
   private get showDashboard(): boolean {
     const {dashboard} = this.props
-    const {servicesStatus} = this.state
-
-    const showDashboard = dashboard && servicesStatus === RemoteDataState.Done
```
Comment (AlirieGray) [inline ui/src/dashboards/containers/DashboardPage.tsx]: what's the reasoning behind getting rid of the loading state check?

File: ui/src/shared/actions/sources.ts [hunk 3]
```diff
@@ -200,14 +203,86 @@ export const deleteKapacitorAsync = (kapacitor: Kapacitor) => async (
...
-    const {
-      data: {sources},
-    } = await getSourcesAJAX()
-    dispatch(loadSources(sources))
-    return sources
+    const {data} = await getSourcesAJAX()
+
+    const sources: Source[] = data.sources
+
+    const sourcesWithFluxLink = await Promise.all(sources.map(addSourceLink))
```
Comment (AlirieGray) [inline ui/src/shared/actions/sources.ts]: very clean!

File: ui/src/shared/actions/sources.ts [hunk 4]
```diff
@@ -200,14 +203,86 @@ export const deleteKapacitorAsync = (kapacitor: Kapacitor) => async (
...
+
+const updateFluxService = async (source: Source, service: Service) => {
+  const {url} = source
+  const updatedService = {...service, url}
+  const s = await updateServiceAJAX(updatedService)
+  return s
+}
+
+const validateSupportsFlux = (source: Source) => {
+  if (source.type !== 'influx') {
```
Comment (AlirieGray) [inline ui/src/shared/actions/sources.ts]: what are the other types for a source?

File: ui/src/shared/actions/sources.ts [hunk 5]
```diff
@@ -200,14 +203,86 @@ export const deleteKapacitorAsync = (kapacitor: Kapacitor) => async (
...
+      } else {
+        if (fluxServices[0].url !== source.url) {
+          service = await updateFluxService(source, fluxServices[0])
+        } else {
+          service = fluxServices[0]
+        }
+      }
+
+      const {links} = source
+      const linksWithFlux: SourceLinks = {...links, flux: service.links.proxy}
```
Comment (AlirieGray) [inline ui/src/shared/actions/sources.ts]: I like this, very readable

File: ui/src/dashboards/containers/DashboardPage.tsx (hunk 2 above)
Comment (ischolten) [reply ui/src/dashboards/containers/DashboardPage.tsx]: its checking whether specifically services are loaded, but we are no longer loading services

File: ui/src/shared/actions/sources.ts (hunk 4 above)
Comment (ischolten) [reply ui/src/shared/actions/sources.ts]: i believe 'influx-enterprise' and maybe more, but we don't want to do this for those.

File: ui/src/dashboards/containers/DashboardPage.tsx (hunk 2 above)
Comment (AlirieGray) [reply ui/src/dashboards/containers/DashboardPage.tsx]: gotcha

**Pull request merged**